wallet-instance-attestation changes (#60)
italia/eidas-it-wallet-docs: docs/en/wallet-instance-attestation.rst

| alg     | Algorithm employed to verify the token signature (e.g., ES256). |
+---------+-----------------------------------------------------------------+
| kid     | Thumbprint of the public key used for signing.                  |
+---------+-----------------------------------------------------------------+
| typ     | Media type, here we use the entity-statement+jwt value.         |
+---------+-----------------------------------------------------------------+

Payload
^^^^^^^
+-----------------------------------+-----------------------------------+
| **Key**                           | **Value**                         |
+-----------------------------------+-----------------------------------+
| iss                               | Public URL of the Wallet          |
|                                   | Provider.                         |
+-----------------------------------+-----------------------------------+
| sub                               | Public URL of the Wallet          |
|                                   | Provider.                         |
+-----------------------------------+-----------------------------------+

> | iat                               | Timestamp of configuration release.|

Reviewer: ```suggestion
| iat                               | Issuance datetime in               |
|                                   |  Unix Timestamp format.            |
```

+-----------------------------------+-----------------------------------+

> | exp                               | Timestamp of configuration expiry. |

Reviewer: ```suggestion
| exp                               | Expiration datetime                |
|                                   | in Unix Timestamp format.          |
```

+------------------------------------+------------------------------------+
| jwks                               | Contains an array of all the Wallet|
|                                    | Provider's public keys.            |
+------------------------------------+------------------------------------+
| token_endpoint                     | Endpoint for obtaining the Wallet  |
|                                    | Instance Attestation.              |
+------------------------------------+------------------------------------+
| asc_values_supported               | List of supported values for the   |
|                                    | certified security context. These  |
|                                    | values specify the security level  |
|                                    | of the app—low, medium, or high.   |
|                                    | An attested security context is    |
|                                    | defined by the proof that the      |
|                                    | Wallet Instance can send to the    |
|                                    | Wallet Provider. Note: this        |

> |                                    | parameter is non-standard and      |

Reviewer 2: parameter is non-standard --> parameter is defined in this specification

Reviewer: Done

Dynamic Component View
~~~~~~~~~~~~~~~~~~~~~~

This section describes the Wallet Instance Attestation format and how the Wallet Provider issues it.

.. figure:: ../../images/dynamic_view_sequence_wallet_instance_attestation.svg
   :name: Sequence Diagram for Wallet Instance Attestation Request
   :alt: The figure illustrates the sequence diagram for issuing a Wallet Instance Attestation, with the steps explained below.
   :target: https://www.plantuml.com/plantuml/uml/XPB1RzKm3CRl-IlCJY3nn7s7QOZ3118IGi0kkxYDLLcqJd2SLMz_FLvV6r7AnDN-_Fi-ExajXcfr6iEhh3XC24Rf2Kmh1QoMf4uTQGZPLTnpHZ6u-bv8hm0Br7tz7iUH33wAGwMdHJBpFpLVD3roN35p5qA5qusBhtsQZN7a9uBvekMLzo19GUbNfMBlib8X1_PAaUHveeIPJpTpTmrtPDjiNdrW8iE8Xc7kJgvoeyzh1VeaXYmimnyqi7EcyXP-qddnPAN9EruXYJcnsEhdf1yUrqbqC3MjnM3aOgxT5hmZ8NNrWix8MhQcH_zwMGyaIK-U5KwNgRNGB3yeFIF-kZYyBuNKE4a3VRh_5h0tVbpoTRiROLE__Y_eZOTP9W_RyZOpa5GM4YhbA2uy25fLQgrXkmDANDe7OClN7ktbXO-FyJ8jqluYpguDtVJSFc9y42MCPx04gJDa0Q5vz_LkIMATnjy0

> - **Message 1**: The User initializes the Wallet Instance. This happens after its installation and each time the Wallet Instance Attestation expires or the User wishes to request or present a credential.

Reviewer 2: Are we really sure that we want to initialize the wallet instance whenever the user wishes to request or present a credential? I know that in this way we always are sure that the Wallet Instance Attestation is fresh, but I see some potential concerns:
1. Reliability: the wallet provider is involved in each issuance and presentation flow, if something wrong with the HTTP request/response with the wallet backend, the issuance/presentation cannot happen. 
2. Privacy: the wallet provider knows some kind of user activity, e.g. the use is going to make a credential request or presentation. 
Just to open a discussion.

Reviewer: see my suggestion below

Reviewer 3: 3. For offline flow this is not possible! How are we going to manage it? Should we have long-term attestation?

> - **Message 1**: The User initializes the Wallet Instance. This happens after its installation and each time the Wallet Instance Attestation expires or the User wishes to request or present a credential.

Reviewer: ```suggestion
- **Message 1**: The User starts the Wallet Instance mobile app, a new Wallet Instance Attestation is automatically obtained if the previous one results expired.
```

+------------------------------------+------------------------------------+
|| grant_types_supported             || The type of grants supported by   |
||                                   || the endpoint token. Therefore,    |
||                                   || for the Wallet Provider the token |
||                                   || is equivalent only to the Wallet  |
||                                   || Instance attestation, therefore   |
||                                   || this attribute will contain an    |
||                                   || array with only one element.      |
+------------------------------------+------------------------------------+
|| token_endpoint_auth_methods_suppo || Supported authentication method   |
|| rted                              || for the endpoint token.           |
||                                   ||                                   |
+------------------------------------+------------------------------------+
|| token_endpoint_auth_signing_alg_v || List of supported signature       |
|| alues_supported                   || algorithms.                       |

> +------------------------------------+------------------------------------+

Reviewer 3: Sorry @peppelinux, but due to the formatting it is difficult to understand what has been modified. Can we find a solution to this problem?

Reviewer: the Wallet Provider Entity Configuration was moved into the wallet solution section as mentioned in the PR description, so it is simple removed from here
